plugin.py

```diff
@@ -31,6 +37,15 @@ def arch() -> str:
         raise RuntimeError("Unknown architecture: " + sublime.arch())
 
 
+def get_setting(view: sublime.View, key: str, default: Optional[Union[str, bool]] = None) -> Union[bool, str]:
+    settings = view.settings()
+    if settings.has(key):
+        return settings.get(key)
+
+    settings = sublime.load_settings('Packages/LSP-rust-analyzer/LSP-rust-analyzer.sublime-settings')
```

Comment:
That was correct. You can revert that part.

What I was saying is that for example `get_setting(view, "rust-analyzer.terminusUsePanel", False)` will read setting from the root level of the `LSP-rust-analyzer.sublime-settings`. Not from the `settings` object where the settings are actually located.

So basically the key should be `rust-analyzer.terminusUsePanel` while reading from the `View` but when reading from the `LSP-rust-analyzer.sublime-settings` it needs to be done differently. You need to either read `settings` object first and then get those settings from the object or use `DottedDict` class and read `settings.rust-analyzer.terminusUsePanel` since DottedDict handles nested objects like that.